qa: Facilitate debugging bitcoind inside functional tests (#31723)
bitcoin/bitcoin: CMakeLists.txt

option(REDUCE_EXPORTS "Attempt to reduce exported symbols in the resulting executables." OFF)
option(WERROR "Treat compiler warnings as errors." OFF)
option(WITH_CCACHE "Attempt to use ccache for compiling." ON)

> option(WAIT_FOR_DEBUGGER "Support for waiting during startup for debugger to be attached." OFF)

Reviewer: ```suggestion
option(WAIT_FOR_DEBUGGER "Pause at startup until a debugger attaches (-waitfordebugger)." OFF)

```
